### PR TITLE
Core timeout fix

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py [hunk 1]
```diff
@@ -227,9 +227,12 @@ def send(self, request, **kwargs): # type: ignore
         try:
             timeout = kwargs.pop('connection_timeout', self.connection_config.timeout)
             if not isinstance(timeout, tuple):
-                _LOGGER.warning("Tuple timeout setting is deprecated")
                 read_timeout = kwargs.pop('read_timeout', self.connection_config.read_timeout)
                 timeout = (timeout, read_timeout)
+            else:
+                if 'read_timeout' in kwargs:
+                    raise ValueError('Conflict, cannot set tuple connection_timeout and read_timeout together')
+                _LOGGER.warning("Tuple timeout setting is deprecated")
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py]: I had to read through this a few time. Small suggestion but I think it's easier to read without the negation in the first branch, and without mutating `timeout` on one branch but not in the other:
```
connection_timeout = kwargs.pop('connection_timeout', self.connection_config.timeout)

if isinstance(connection_timeout, tuple):
    if 'read_timeout' in kwargs:
        raise ValueError('Cannot set tuple connection_timeout and read_timeout together')
    _LOGGER.warning("Tuple timeout setting is deprecated")
    timeout = connection_timeout
else:
    read_timeout = kwargs.pop('read_timeout', self.connection_config.read_timeout)
    timeout = (connection_timeout, read_timeout)
```

File: sdk/core/azure-core/tests/test_basic_transport.py [hunk 2]
```diff
@@ -388,4 +388,24 @@ def body(self):
...
+def test_timeout(caplog):
+    transport = RequestsTransport()
+
+    request = HttpRequest("GET", "https://www.bing.com")
+
+    with caplog.at_level(logging.WARNING, logger="azure.core.pipeline.transport"):
+        with Pipeline(transport) as pipeline:
+            pipeline.run(request, connection_timeout=100)
+
+    assert "Tuple timeout setting is deprecated" not in caplog.text
```
Comment (bryevdv) [inline sdk/core/azure-core/tests/test_basic_transport.py]: Suggest positive test for case where the deprecation is raised

**Comment by bryevdv** [comment]
LGTM assuming green :)

**Pull request merged**